diff --git a/src/platformtheme/kdeplatformfiledialoghelper.cpp b/src/platformtheme/kdeplatformfiledialoghelper.cpp
index 02aa56e..194846f 100644
--- a/src/platformtheme/kdeplatformfiledialoghelper.cpp
+++ b/src/platformtheme/kdeplatformfiledialoghelper.cpp
@@ -221,8 +221,14 @@ KDEPlatformFileDialogHelper::~KDEPlatformFileDialogHelper()
 void KDEPlatformFileDialogHelper::initializeDialog()
 {
     if (options()->testOption(QFileDialogOptions::ShowDirsOnly)) {
-        delete m_dialog;
-        m_dialog = new KDirSelectDialog(options()->initialDirectory());
+        m_dialog->deleteLater();
+        m_dialog = new KDirSelectDialog(
+#if QT_VERSION >= QT_VERSION_CHECK(5,5,0)
+            options()->initialDirectory()
+#else
+            m_dialog->directory()
+#endif
+        );
         connect(m_dialog, SIGNAL(accepted()), SIGNAL(accept()));
         connect(m_dialog, SIGNAL(rejected()), SIGNAL(reject()));
     } else {
@@ -234,7 +240,9 @@ void KDEPlatformFileDialogHelper::initializeDialog()
         } else {
             dialog->setWindowTitle(options()->windowTitle());
         }
+#if QT_VERSION >= QT_VERSION_CHECK(5,5,0)
         setDirectory(options()->initialDirectory());
+#endif
         //dialog->setViewMode(options()->viewMode()); // don't override our options, fixes remembering the chosen view mode and sizes!
         dialog->setFileMode(options()->fileMode());
 
